qiskit/circuit/instruction.py

```diff
@@ -96,8 +96,19 @@ def __eq__(self, other):
             return False
 
         for self_param, other_param in zip_longest(self.params, other.params):
-            if self_param == other_param:
-                continue
+            try:
+                if self_param == other_param:
+                    continue
+            except ValueError:
+                pass
+
+            try:
+                if numpy.shape(self_param) == numpy.shape(other_param) \
+                   and numpy.allclose(self_param, other_param,
```

Comment:
for array params we are using approximate comparison, but for single params it is using exact comparison. probably should be consistent?

Reply:
Currently, for scalar parameters, we try exact comparison first, and then try an approximate comparison if the exact comparison fails.

I can add this same pattern for arrays if we think it makes sense, but I'm not sure I see when it would be applied (array-like parameters of `sympy.Symbol`s, maybe?)